woocommerce/woocommerce
# Change blocks registering logic in widget area from opt-out to opt-in

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This pull request includes updating the block registration strategy for Widget Areas from an opt-out to an opt-in approach.

That required going through all the blocks and choosing ones that gets registered in Widget area. I went through BlockTypes list:

https://github.com/woocommerce/woocommerce/blob/3f6b19126e26235ba6df5688632e96f8be6a08f5/plugins/woocommerce/src/Blocks/BlockTypesController.php#L415

and chose those that work in widget area and makes sense there.

1. Keep in mind I kept product list blocks:

```
			'HandpickedProducts',
			'ProductBestSellers',
			'ProductNew',
			'ProductOnSale',
			'ProductTopRated',
			'ProductsByAttribute',
			'ProductCategory',
			'ProductTag',
```

They are hidden from inserter but they work correctly in this area hence keeping them for backward compatibility in case someone displays products in that way.

2. I wasn't sure what's `PageContentWrapper` so kept it to be on a safe side of things.

---
Comment regarding changes in `plugins/woocommerce/tests/e2e-pw/tests/customize-store/intro.spec.js`:

One E2E test was failing with these changes. Not sure why but I checked and seems like `await page.waitForNavigation();` wasn't needed at all. Maybe the current logic is slightly faster/slower causing some race condition, not sure 😬 

Closes https://github.com/woocommerce/woocommerce/issues/53691



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Switch to classic theme, for example Storefront
2. Go to Appearance > Widgets
4. Try adding blocks in Sidebar area:
    1. That is no longer available in the area, for example: Single Product.
    2. **Expected:** Block IS NOT available in the inserter.
    3. That is still available in the area, for example: Cart Link or Product Search.
    5. **Expected:** Block IS available in the inserter.



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
